Doc/using/cmdline.rst

```diff
@@ -302,16 +302,22 @@ Miscellaneous options
 
 .. option:: -i
 
-   Enter interactive mode after execution.
+   Enter interactive mode after execution, or force interactive mode even when
+   :data:`sys.stdin` does not appear to be a terminal.
 
    Using the :option:`-i` option will enter interactive mode in any of the following circumstances\:
 
    * When a script is passed as first argument
    * When the :option:`-c` option is used
    * When the :option:`-m` option is used
 
-   Interactive mode will start even when :data:`sys.stdin` does not appear to be a terminal. The
-   :envvar:`PYTHONSTARTUP` file is not read.
+   In these "execute then interact" cases, Python runs the script or command
+   first and does not read the :envvar:`PYTHONSTARTUP` file before entering
+   interactive mode.
```

Comment:
I don't understand this paragraph (it looks wrong). For example, `PYTHONSTARTUP=script.py ./python -i -c pass` **does** execute `script.py`.